### PR TITLE
validate against ambiguous references

### PR DESCRIPTION
**Pull request by carterkozak**
## After this PR
==COMMIT_MSG==

==COMMIT_MSG==

### CODE REVIEW

File: baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/LambdaMethodReference.java [hunk 1]
```diff
@@ -184,12 +185,56 @@ private Description convertMethodInvocations(
...
+            Symbol.MethodSymbol symbol, @Nullable ExpressionTree receiver, VisitorState state) {
+        if (symbol.isStatic()) {
+            if (symbol.params().size() != 1) {
+                return false;
+            }
+            Symbol.ClassSymbol classSymbol = ASTHelpers.enclosingClass(symbol);
+            if (classSymbol == null) {
+                return false;
+            }
+            Set<Symbol.MethodSymbol> matching = ASTHelpers.findMatchingMethods(
```
Comment (ferozco) [inline baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/LambdaMethodReference.java:210]: this is a super cool helper method! didn't know it existed

File: baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/LambdaMethodReferenceTest.java [hunk 2]
```diff
@@ -23,23 +23,21 @@
...
 import java.util.Optional;
 import java.util.function.Supplier;
-import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 
 public class LambdaMethodReferenceTest {
 
-    private CompilationTestHelper compilationHelper;
-    private RefactoringValidator refactoringValidator;
+    private CompilationTestHelper compile() {
```
Comment (ferozco) [inline baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/LambdaMethodReferenceTest.java:30]: gotta keep those tests fast 😛

**Review by ferozco — APPROVED** [review]

File: baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/LambdaMethodReferenceTest.java (hunk 2 above)
Comment (carterkozak) [reply baseline-error-prone/src/test/java/com/palantir/baseline/errorprone/LambdaMethodReferenceTest.java:30]: heh, most of our tests are structured this way so I moved it over. I think I migrated most of them after hitting a few cases where tests flaked after helper/validator instances were reused (due to either parallelism or multiple asserts in a single test). This way they're always clean regardless of other state!

**Comment by carterkozak** [comment]
There's one more bug that I'm aware of where we trasform `foo()` to `this.foo()` regardless of whether the enclosing class (`this`) provides `foo()`. This breaks when `foo()` exists on an enclosing class, should be simple to check though.

**Pull request merged**